### PR TITLE
CVS-58781 Scale/mean preprocessing example server side with custom node

### PR DESCRIPTION
**Pull request by dkalinowski**

### CODE REVIEW

**Review by rasapala — APPROVED** [review]
LGTM

File: docs/ovms_onnx_example.md [hunk 1]
```diff
@@ -13,13 +13,10 @@ resnet/
 └── 1
     └── resnet50-caffe2-v1-9.onnx
 ```
-Note that the downloaded model requires additional [preprocessing function](https://github.com/onnx/models/tree/master/vision/classification/resnet#preprocessing).  
+Note that the downloaded model requires additional [preprocessing function](https://github.com/onnx/models/tree/master/vision/classification/resnet#preprocessing). Preprocessing can be performed in the client by manipulating data before sending the request. Preprocessing can be also done on the server side by creating DAG and using custom processing node. Both methods will be explained below.
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:16]:
```suggestion
Note that the downloaded model requires additional [preprocessing function](https://github.com/onnx/models/tree/master/vision/classification/resnet#preprocessing). Preprocessing can be performed in the client by manipulating data before sending the request. Preprocessing can be also delegated to the server by creating a DAG and using a custom processing node. Both methods will be explained below.
```

File: docs/ovms_onnx_example.md [hunk 2]
```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
...
+├── config_with_preprocessing_node.json
+├── lib
+│   └── libcustom_node_image_transformation.so
+└── models
+    └── resnet50-caffe2-v1
+        └── 1
+            └── resnet50-caffe2-v1-9.onnx
+```
+
+Start the OVMS container with configuration file option:
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:118]:
```suggestion
Start the OVMS container with a configuration file option:
```

File: docs/ovms_onnx_example.md [hunk 3]
```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
...
+```bash
+docker run -d -u $(id -u):$(id -g) -v $(pwd)/resnet:/model -p 9001:9001 openvino/model_server:latest \
+--config_path /workspace/config_with_preprocessing_node.json --port 9001
+```
+
+Use example binary input client to send JPEG images to the server:
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:124]:
```suggestion
Use a sample client to send JPEG images to the server:
```

File: docs/ovms_onnx_example.md [hunk 4]
```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
...
+```
+$ cd model_server/example_client
+
+$ python3 grpc_binary_client.py --grpc_port 9001 --input_name 0 --output_name 1463 --model_name resnet --batchsize 1
+```
+You should see 100% overall accurracy:
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:130]:
```suggestion
Below is the client output including performance and accuracy results:
```

File: docs/ovms_onnx_example.md [hunk 5]
```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
...
+         8 pelican 144 ; Correct match.
+Batch: 8; Processing time: 41.93 ms; speed 23.85 fps
+         9 snail 113 ; Correct match.
+Batch: 9; Processing time: 38.24 ms; speed 26.15 fps
+         10 zebra 340 ; Correct match.
+Overall accuracy= 100.0 %
+Average latency= 37.2 ms
+```
+
+## Summary
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:159]:
```suggestion
## Node parameters explanation
```

File: docs/ovms_onnx_example.md [hunk 6]
```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
...
+Additional preprocessing step allowed us to apply division and substraction to each pixel value in the image. We performed this calculation by passing two parameters to _image transformation_ custom node:
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:160]:
```suggestion
Additional preprocessing step applies a division and an subtraction to each pixel value in the image. This calculation is configured by passing two parameters to _image transformation_ custom node:
```

File: docs/ovms_onnx_example.md [hunk 7]
```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
...
+```
+"params": {
+  ...
+  "mean_values": "[123.675,116.28,103.53]",
+  "scale_values": "[58.395,57.12,57.375]",
+  ...
+}
+```
+For each pixel we substracted `123.675` from blue value, `116.28` from green value and `103.53` from red value. Next, we performed division operation in the same color order using `58.395`, `57.12`, `57.375` values. This way we matched image data to input required by onnx model.
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:169]:
```suggestion
For each pixel, the custom node subtractes `123.675` from blue value, `116.28` from green value and `103.53` from red value. Next, it divides in the same color order using `58.395`, `57.12`, `57.375` values. This way we match the image data to the input required by onnx model.
```

File: docs/ovms_onnx_example.md [hunk 8]
```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
...
+$ cd model_server/example_client
+
+$ python3 grpc_binary_client.py --grpc_port 9001 --input_name 0 --output_name 1463 --model_name resnet --batchsize 1
+```
+You should see 100% overall accurracy:
+```
+Start processing:
+        Model name: resnet
+        Images list file: input_images.txt
+Batch: 0; Processing time: 38.65 ms; speed 25.87 fps
```
Comment (dtrawins) [inline docs/ovms_onnx_example.md:135]: maybe the performance would be better with 1 stream on ovms side?

Comment (dkalinowski) [reply docs/ovms_onnx_example.md]: There is significant perf boosst with 1 stream. Changed the doc and config file.

**Review by dtrawins — APPROVED** [review]

**Pull request merged**